routes/openai_route.py

```diff
@@ -1,18 +1,48 @@
 from fastapi import APIRouter, Response
-from services.openai_service import get_chat_completion, get_question_and_answer
-from data_types.openai_req_types import GenQuestionBody as GenBody
+from globals import globals
+from data_access_layer import users
+from services.openai_service import get_question_and_answer,evaluate_answer
+from data_types.openai_req_types import GenQuestionBody as GenBody, QARequest
+from pymongo.errors import PyMongoError
 
 router = APIRouter()
 
+
 @router.post('/generate')
-async def gen_question(body:GenBody, response: Response):
+async def gen_question(body: GenBody, response: Response):
     topic = body.topic
     difficulty = body.difficulty
     answers_num = body.answers_num
     try:
-        answer = get_question_and_answer(topic,difficulty,answers_num)
+        answer = get_question_and_answer(topic, difficulty, answers_num)
         return answer
     except Exception as e:
         print(e)
         response.status_code = 400
-        return e
\ No newline at end of file
+        return e
+
+
+
+@router.post('/evaluate')
+async def evaluate_question(body: QARequest, response: Response):
+    try:
+        user_id = body.user_id
+        question_text = body.question_text
+        topic = body.topic
+        difficulty = body.difficulty
+        answer = body.answer
+        evaluation_score = evaluate_answer(question=question_text, answer=answer)
+        users.add_user_stats(user_id=user_id, question_text=question_text, answer=answer, topic=topic,
+                             difficulty=difficulty,
+                             score=evaluation_score["Score"], answer_correct=(evaluation_score["Score"] >= 5),
+                             client=globals.mongo_client)
+        return evaluation_score
```

Comment:
You must return the `score and the explanation with the question and user answer` but you return the score only, please consider changing this

Reply:
its already returning the explanation , I added question and answer